app/commands/git/pull_request/sync_repo.rb

```diff
@@ -0,0 +1,146 @@
+        loop do
+          page_data = fetch_page(cursor)
+
+          # TODO: filter out PRs we want to ignore (e.g. the v3 bulk rename PRs)
+          results += pull_requests_from_page_data(page_data)
+
+          break results unless page_data[:data][:repository][:pullRequests][:pageInfo][:hasNextPage]
+
+          cursor = page_data[:data][:repository][:pullRequests][:pageInfo][:endCursor]
+
+          handle_rate_limit(page_data[:data][:rateLimit])
+        end
+      end
+
+      def fetch_page(cursor)
+        query = <<~QUERY.strip
+            repository(owner: "#{repo_owner}", name: "#{repo_name}") {
+              nameWithOwner
+              pullRequests(first: 100,
+                          states:[CLOSED, MERGED]
+                          #{cursor ? ", after: \"#{cursor}\"" : ''}) {
+                nodes {
+                  id
+                  url
+                  title#{'                  '}
+                  number
+                  merged
+                  mergedBy {
+                    login
+                  }
+                  labels(first: 100) {
+                    nodes {
+                      name
+                    }
+                  }
+                  author {
+                    login
+                  }
+                  reviews(first: 100) {
+                    nodes {
+                      id
+                      author {
+                        login
+                      }
+                    }
+                  }
+                }
+                pageInfo {
+                  hasNextPage
+                  endCursor
+                }
+              }
+            }
+            rateLimit {
+              remaining
+              resetAt
+            }
+          }
+        QUERY
+
+        octokit_client.post("https://api.github.com/graphql", { query: query }.to_json)
+      end
+
+      def pull_requests_from_page_data(response)
+        # We're transforming the GraphQL response to the format used to call
+        # User::ReputationToken::AwardForPullRequest that is called when the
+        # pull request update webhook fires.
+        # This allows us to work with pull requests using a single code path.
+        response[:data][:repository][:pullRequests][:nodes].map do |pr|
+          next if pr[:author].nil? # In rare cases the PR author is null
+
+          {
+            action: 'closed',
+            author: pr[:author][:login],
+            url: "https://api.github.com/repos/#{response[:data][:repository][:nameWithOwner]}/pulls/#{pr[:number]}",
+            html_url: pr[:url],
+            labels: pr[:labels][:nodes].map { |node| node[:name] },
+            state: 'closed',
+            pr_id: pr[:id],
+            pr_number: pr[:number],
+            repo: response[:data][:repository][:nameWithOwner],
+            merged: pr[:merged],
+            merged_by: pr[:mergedBy].present? ? pr[:mergedBy][:login] : nil,
+            reviews: pr[:reviews][:nodes].map do |node|
```

Comment:
You can use `filter_map` for this I think and then drop the compact.

Reply:
I've change the import a bit to allow importing pull requests where the author is nil, as we still need the pull request for the reviews/merger reputation.